Library/Homebrew/os/mac/sdk.rb

```diff
@@ -93,10 +93,21 @@ def sdk_paths
 
             # Use unversioned SDK path on Big Sur to avoid issues such as:
             # https://github.com/Homebrew/homebrew-core/issues/67075
-            if OS::Mac.version >= :big_sur
-              sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
-              version = OS::Mac.full_version
-              paths[version] = sdk_path if File.directory?(sdk_path)
+            sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
+            if OS::Mac.version >= :big_sur && File.directory?(sdk_path)
+              sdk_settings_json = File.join(sdk_path, "SDKSettings.json")
```

Comment:
Might be worth pulling all the JSON/Plist reading into a separate, private function?

Reply:
Hmm, I'm not sure. We're only reading twice, and only one of each kind of file.

Reply:
I'll look again once it's only the JSON stuff.